src/Adapter/Product/Repository/CombinationRepository.php

```diff
@@ -0,0 +1,253 @@
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Repository;
+
+use Combination;
+use Db;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Adapter\AbstractObjectModelRepository;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CannotAddCombinationException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CombinationConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShopException;
+
+/**
+ * Provides access to Combination data source
+ */
+class CombinationRepository extends AbstractObjectModelRepository
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+    }
+
+    /**
+     * @param Combination $combination
+     *
+     * @return CombinationId
+     *
+     * @throws CombinationConstraintException
+     * @throws CoreException
+     */
+    public function add(Combination $combination): CombinationId
+    {
+        //@todo: validation?
+        $id = $this->addObjectModel($combination, CannotAddCombinationException::class);
+
+        return new CombinationId($id);
+    }
+
+    /**
+     * @param int $productId
+     * @param int $limit
+     * @param int $offset
+     * @param array $filters
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    public function getProductCombinations(int $productId, ?int $limit = null, ?int $offset = null, array $filters = []): array
+    {
+        $qb = $this->getCombinationsQueryBuilder($productId, $filters)
```

Comment:
As commented in the `GetProductCombinationsForEditingHandler`, this should check for product existence with `ProductRepository:assertProductExists`
Also I wonder if it should use `ProductId` as an input (since the handler uses a VO) or an `int` 🤷‍♂️